echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/AbstractEventNotificationAgent.groovy

```diff
@@ -78,7 +78,13 @@ abstract class AbstractEventNotificationAgent implements EchoEventListener {
       return
     }
 
-    if (config.type == 'pipeline' && event.content.execution?.status == 'CANCELED') {
+    if (config.type == 'stage' && event.content.canceled == true) {
```

Comment:
so one repro/failure scenario is, have a "failure" notification on a stage, cancel a pipeline while that stage is running?

I don't understand the second repro case (below)?

Reply:
oh, i see.... when we cancel the execution, we don't refetch the execution from the db hence `canceled = true` but `status` is probably still `= running`..? but I still don't understand why this doesn't happen always..?

Reply:
Yes, that’s what I’m thinking, but I couldn’t quite pin it down, so this was more of a hunch...